tests/apps/test_graph_utils.py

```diff
@@ -186,3 +186,91 @@ def test_invalid_subgraphs(self, graph, dim):
         subgraphs = itertools.combinations(range(dim), int(dim / 2))
 
         assert not all([utils.is_subgraph(list(s), graph) for s in subgraphs])
+
+
+@pytest.mark.parametrize("dim", range(2, 10))
+class TestIsClique:
+    """Tests for the function `strawberryfields.apps.graph.utils.is_clique` """
+
+    def test_no_false_negatives(self, dim):
+        """ Tests that cliques are labelled as such"""
+        g = nx.complete_graph(dim)
+        assert utils.is_clique(g)
+
+    def test_no_false_positives(self, dim):
+        """ Tests that non-cliques are labelled as such"""
+        g = nx.empty_graph(dim)
+        assert not utils.is_clique(g)
+
+
+@pytest.mark.parametrize("dim", range(2, 10))
+class TestC0:
+    """ Tests function c_0 that generates the set of nodes connected to all nodes in a clique"""
+
+    def test_grows_to_clique(self, dim):
```

Comment:
I'm not sure about this test, as I think it's more of a remake in test form of the `clique_grow` function we plan. The test implicitly relies on the `c_0` function but there could also be mistakes in the test (you mentioned tests of tests :grin:). Of course it's a balance, and the code in this test isn't too complicated.

However, the `test_c0_comp_graph` is more along the lines of what I was thinking. How about something like:
```python
@pytest.mark.parametrize("subdim", range(2, 10))
def test_correct_c0(self, dim, subdim):
    g = nx.lollipop_graph(dim, 1)
    s = set(range(subdim))
    res = set(c_0(s, g))

    assert res not in s
    assert {dim + 1} not in res
    assert res | s == set(range(dim))
```

Reply:
I agree here - the unit test should be testing the existing functionality of the c0 function, and it sounds like this test is a future unit test of `clique_grow`.

But you might as well keep it here for now, until the PR that introduces `clique_grow`. You could mark it with a `# TODO:` so that it is not forgotten.

Reply:
Did we disable `TODO` checks in pylint?